src/schema/bookmarks.ts

```diff
@@ -300,22 +307,39 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       source,
       { data }: { data: GQLAddBookmarkInput },
       ctx: AuthContext,
-    ): Promise<GQLEmptyResponse> => {
+      info,
+    ): Promise<GQLBookmark[]> => {
+      const lastAddedBookmark = await ctx.con.getRepository(Bookmark).findOne({
+        where: { userId: ctx.userId },
+        select: ['listId'],
+      });
+      const lastUsedListId = lastAddedBookmark?.listId ?? null;
+
       const [query, params] = ctx.con
         .createQueryBuilder()
         .select('id', 'postId')
         .addSelect(`'${ctx.userId}'`, 'userId')
+        .addSelect(`:listId`, 'listId')
         .from(Post, 'post')
         .where('post.id IN (:...postIds)', { postIds: data.postIds })
+        .setParameter('listId', lastUsedListId)
         .getQueryAndParameters();
+
       await ctx.con.query(
-        `insert into bookmark("postId", "userId") ${query} on conflict do nothing`,
+        `insert into bookmark("postId", "userId", "listId") ${query} on conflict do nothing`,
         params,
       );
-      return { _: true };
+
+      return await graphorm.query<GQLBookmark>(ctx, info, (builder) => ({
```

Comment:
Why do we need the return data here again?
Just because we don't know where it's going to be. moved to?

Reply:
Agree with Chris, can we just return `listId` if its needed, we should be able to do:
```suggestion
     return {
        lastUsedListId
     }
```

Reply:
The reason why we would want to only return stuff we need or can compute is because of read replica. If we query bookmark after addition we need to query on master which is worse and adds another read to the master db.

Reply:
@capJavert @rebelchris We need the bookmark just added because on frontend we can change folder using toast action button. So we need the list.name and bookmark object for referencing it 👀

Reply:
you know post ID on the frontend?

Reply:
@ilasw we can merge this to unblock but let's look on frontend side and try to avoid querying everything with regards to above comments. Then we should update this so it returns the computed data.